### ISSUE TITLE
Add support for Spotify playlist management

### ISSUE THREAD

**Issue by jodal** [mopidy/mopidy-spotify#22]
Moved from mopidy/mopidy#42:

> It would be nice to be able to manage Spotify playlists through Mopidy. E.g. create/rename/remove playlists, and add/move/remove tracks in playlists.

**Comment by peterlvilim** [comment]
Was thinking of adding this feature.
- Create new spotify playlists
- Add and remove tracks from spotify playlists
- Rename spotify playlists

Right now I assume any playlist that you create would have to only include spotify items in order for it to sync to spotify.  I'm guessing this feature might also require additions to pyspotify which I could do as needed.

Planning to read through some of the code at which point I'll have something more intelligent to say on this topic.

**Comment by charles-l** [comment]
:+1:

**Comment by lino** [comment]
#80 should do the trick.

**Comment by Cysioland** [comment]
Currently getting 'Playlist with scheme "spotify" can't add track with scheme "spotify"' error in ncmpcpp, which does not make sense.

**Comment by jodal** [comment]
Please don't hijack unrelated issues.

**Comment by Cysioland** [comment]
But it is related, because Mopidy-Spotify currently can't manage playlists,
hence this message.

**Comment by pbogut** [comment]
Any update about this one?

**Comment by yannik-b** [comment]
Not now, spotify will close libspotify, which is used by this project, until 2017 or it is closed already (https://github.com/mopidy/mopidy-spotify/issues/110)
So there will (proprably) be currently no changes for libspotify.

**Comment by kingosticks** [comment]
There are no changes required to libspotify (or pyspotify) to implement this feature as they already have [full support for modifying playlists](https://pyspotify.mopidy.com/en/latest/api/playlist/). The changes required to implement this feature are in mopidy-spotify/mopidy. 

If I remember correctly the main blocker is some sensible generic handling when tracks come from multiple backends, as per @peterlvilim's comment above.

So, no update here sadly.

**Comment by Radvendii** [comment]
I think ideally you could add tracks from multiple sources to a playlist in mopidy, and it would sync only those tracks that are from spotify back to the spotify playlist, no?

**Comment by ecly** [comment]
Is there anything going on with this?

**Comment by kingosticks** [comment]
I don't know of anyone working on it.

We need to start using the Web API for most playlist stuff so that probably
includes this. Although since you can't edit the Spotify owned playlists
(the ones seemingly now incompatible with libspotify) that might not
technically be the case. But doing everything the same way does have value.

**Comment by ecly** [comment]
> Although since you can't edit the Spotify owned playlists...

To understand that correctly - does that mean that as things are right now, it will never be possible to add tracks to a playlist that was created directly in Spotify? As in have playlist changes made both in Spotify and with Mopidy be synced.

**Comment by kingosticks** [comment]
No, I'm referring to Spotify playlists owned by the Spotify account. Such
as toplists, new releases etc.

**Comment by ecly** [comment]
Ah of course - alright, so eventual support for editing one's own spotify playlists should be possible then.
